src/main/kotlin/me/moeszyslak/polly/services/MacroService.kt

```diff
@@ -85,17 +136,63 @@ class MacroService(private val store: MacroStore, private val discord: Discord)
         val name = name_raw.toLowerCase()
         val category = category_raw.toLowerCase()
 
-        val result = store.forGuild(guild) {
-            if (it.containsKey("$name#$channelId")) {
-                it["$name#$channelId"]!!.category = category
-                true
-            } else {
-                false
+        val macro = store.forGuild(guild) { macros ->
+            macros["$name#$channelId"]?.let {
+                it.category = category
+                return@forGuild it
             }
+            return@forGuild null
         }
 
-        return if (result) {
-            "Success. Macro `$name` available ${if (channel == null) "globally" else "on channel ${channel.mention}"} is now in category `${category}`"
+        return if (macro != null) {
+            val names = macro.displayNames()
+            "Success. Macro $names available ${if (channel == null) "globally" else "in channel ${channel.mention}"} is now in category `${category}`"
+        } else {
+            "Cannot find a macro by that name. If it is a channel specific macro you need to provide the channel as well."
+        }
+    }
+
+    fun addMacroAlias(guild: GuildId, name_raw: String, channel: TextChannel?, alias_raw: String): String {
+        val channelId = channel?.id?.value ?: ""
+        val name = name_raw.toLowerCase()
+        val alias = alias_raw.toLowerCase()
+
+       store.withAliases(guild) {
+            it["$alias#$channelId"]
+        }?.let {
```

Comment:
optional: can't we just use an if-statement?
Simplicity trumps trying to use fancy features